### ISSUE TITLE
isMobile works on localhost, but not when pushed to the server

### ISSUE THREAD

**Issue by pnbkn** [duskload/react-device-detect#142]
I am using Gatsby and everything works fine on localhost. As soon as I deploy it to the server, it doesn't work.

**Comment by nresh** [comment]
@pnbkn I was having a similar issue.  I've been working on a gatsby app and when I would run `gatsby develop`, both `isMobile` and `<MobileView>` would work as expected.

however, when I deployed to prod, or ran the built version locally by running `gatsby build` and `gatsby serve`, it would not work and seemingly not recognize mobile devices.

after doing some console logging, I found that `isMobile` was printing as `false` in my terminal when the static html pages were being generated during the build process, but the same variables were printing correctly as `true` in my browser console when I inspected the page as a mobile device in my browser.

so, it seems that when serving the built version of the site, `isMobile` was being evaluated too early, before a mobile device could be detected.  on the other hand, since the local version of the site that loads on localhost thru `gatsby develop` is designed for debugging/live reloading it correctly calculates the value of `isMobile` on the client side when the browser loads the home page, and so it was working there correctly.

for me, the solution was to use the react `useEffect()` hook to detect changes to `isMobile` on prod and change the layout as desired. 

so, in my index.js landing page instead of:
```
return (
    <>
        <BrowserView> ... </BrowserView>
        <MobileView> ... </MobileView>
    </>
  )
 ```
 
 I did the following:
 ```
const [layout, setLayout] = useState(<></>);

const mobileLayout = <MyMobileLayout> ... </MyMobileLayout>;
const desktopLayout = <MyDesktopLayout> ... </MyDesktopLayout>;

useEffect(() => {
    setLayout(isMobile ? mobileLayout : desktopLayout)
}, [isMobile, mobileLayout, desktopLayout]);

return (
    <> {layout} </>
 )
 ```
